### PR TITLE
completed the product card task

### PR DESCRIPTION
**Pull request by Shevtsooov**
- [DEMO LINK](https://Shevtsooov.github.io/layout_product-cards/)
- [TEST REPORT LINK](https://Shevtsooov.github.io/layout_product-cards/report/html_report/)

### CODE REVIEW

**Review by snuber-w — APPROVED** [review]

**Review by niaYaro — COMMENTED** [review]
God job!
I suppose you can try to correct your test results by adding to .card__iMacImage display: block and changing margin-bottom: 40px to margin: 0 auto 40px

File: src/index.html [hunk 1]
```diff
@@ -1,13 +1,64 @@
...
+      data-qa="card"
+    >
+      <img
+        src="./images/imac.jpeg"
+        alt="APPLE A1419 iMac 27&quot; Retina 5K Monoblock (MNED2UA/A)"
+        class="card__iMacImage"
+      >
+      <h2 class="card__itemName">
+        APPLE A1419 iMac 27" Retina 5K Monoblock (MNED2UA/A)
+      </h2>
```
Comment (Dima-Petrov) [inline src/index.html:34]: I prefer to use div or span here for much more options of styling. h tags push you to use resets. it's up to you, not a mistake.

File: src/index.html [hunk 2]
```diff
@@ -1,13 +1,64 @@
...
+      <p class="card__productCode">Product code: 195434</p>
```
Comment (Dima-Petrov) [inline src/index.html:35]: u can wrap Product code: and 195434 in divs ad flex them.

File: src/styles/style.css [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+  /* button bg and :hover border+text color */
+  --button-bg-color: #00acdc;
+
+  --button-text-color: #fff;
+}
+
+* {
+  box-sizing: border-box;
+}
```
Comment (Dima-Petrov) [inline src/styles/style.css:15]: i'd rather put these styles in another file like normalize.css. this is why my own task was rejected

File: src/styles/style.css [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+h2,
+p {
+  margin: 0;
+}
+
+.card {
+  width: 200px;
+  padding: 32px 16px 16px;
```
Comment (Dima-Petrov) [inline src/styles/style.css:29]: you should not have any paddings in this task. it is why you don't pass tests. just play with margins and flex.

File: src/styles/style.css [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+  --button-text-color: #fff;
+}
+
+* {
+  box-sizing: border-box;
+}
+
+body {
+  margin: 0;
```
Comment (Dima-Petrov) [inline src/styles/style.css:18]: reset.css

File: src/index.html [hunk 6]
```diff
@@ -1,13 +1,64 @@
...
+          <div class="stars__star"></div>
+
+          <div class="stars__star"></div>
+        </div>
+        <span>Reviews: 5</span>
+      </section>
+      <div class="card__priceContainer">
+        <p class="priceText">Price:</p>
+
+        <p class="priceTag">$2,199</p>
```
Comment (Dima-Petrov) [inline src/index.html:53]: divs + flex

File: src/styles/style.css [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+.card__buyButton {
+  display: block;
+
+  font-size: 14px;
+  height: 40px;
+  font-weight: 700;
+  text-align: center;
+  text-decoration: none;
```
Comment (Dima-Petrov) [inline src/styles/style.css:109]: a tag in reset.css for example

File: src/styles/style.css [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+  margin-bottom: 16px;
+
+  font-size: 12px;
+  font-weight: 400;
+  line-height: 18px;
+  color: var(--secondary-text-color);
+}
+
+.priceTag {
+  margin-bottom: 16px;
```
Comment (Dima-Petrov) [inline src/styles/style.css:94]: why not use margins for the parent container?

**Review by Dima-Petrov — CHANGES_REQUESTED** [review]
GJ Mate:) Have some recommendations from me:
1. Try not to use camelcase in class names. Pretty tough to read + your one-word classes look weird.
2. I highly recommend you try <div>Text</div> or spans. It gives you so many options for positioning and styling.
3. Don't use padding here.
4. Create more files like reset or normalize.css and put there your default values. Create a file with imports and connect only this 1 file to your code.
5. Check the names of your connected styles files. There should be no other "style.css". Pretty confusing.

File: src/index.html (hunk 1 above)
Comment (Shevtsooov) [reply src/index.html:34]: I am good with h2 tag here.

File: src/index.html (hunk 2 above)
Comment (Shevtsooov) [reply src/index.html:35]: Correct, but `p` tag works well here as well.

File: src/styles/style.css (hunk 3 above)
Comment (Shevtsooov) [reply src/styles/style.css:15]: Strongly agree, will do that!
Thanks!

File: src/styles/style.css (hunk 4 above)
Comment (Shevtsooov) [reply src/styles/style.css:29]: The test has been passed.
(There are some more commits after the PR itself, so I have resolved it already).

The test failed due to absence of `display:block` in the `img` tag.

File: src/styles/style.css (hunk 5 above)
Comment (Shevtsooov) [reply src/styles/style.css:18]: Sure, will create another file for that.

File: src/index.html (hunk 6 above)
Comment (Shevtsooov) [reply src/index.html:53]: There is `div `which has `flex`, so it does the job.

File: src/styles/style.css (hunk 7 above)
Comment (Shevtsooov) [reply src/styles/style.css:109]: Will do!

File: src/styles/style.css (hunk 8 above)
Comment (Shevtsooov) [reply src/styles/style.css:94]: We were also allowed to use margins in this way.

File: src/styles/style.css [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+
+  font-size: 16px;
+  font-weight: 700;
+  text-align: end;
```
[the remark this answers is not in the document]
Comment (Shevtsooov) [reply src/styles/style.css:98]: I just forgot to remove this line.
I used it as the workaround for the first time.

Will remove it!

File: src/styles/style.css (hunk 4 above)
Comment (Shevtsooov) [reply src/styles/style.css:29]: The test has been passed.
(There are some more commits after the PR itself, so I have resolved it already).

The test failed due to absence of display:block in the img tag.

I just forgot to deploy the new version :(

**Comment by Shevtsooov** [comment]
@Dima-Petrov Thank you for the useful review!
The changes have been applied.

File: src/index.html [hunk 10]
```diff
@@ -1,13 +1,69 @@
...
+  </head>
+  <body>
+    <section
+      class="card"
+      data-qa="card"
+    >
+      <img
+        src="./images/imac.jpeg"
+        alt="APPLE A1419 iMac 27&quot; Retina 5K Monoblock (MNED2UA/A)"
+        class="card__iMac-image"
```
Comment (silchyn) [inline src/index.html:30]:
```suggestion
        class="card__image"
```

File: src/index.html [hunk 11]
```diff
@@ -1,13 +1,69 @@
...
+
+          <div class="stars__star"></div>
+
+          <div class="stars__star"></div>
+
+          <div class="stars__star"></div>
+
+          <div class="stars__star"></div>
+        </div>
+        <span>Reviews: 5</span>
```
Comment (silchyn) [inline src/index.html:51]: checklist `#12`

File: src/styles/card.css [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+  display: block;
+
+  margin: 0 auto 40px;
+
+  width: 160px;
+  height: 134px;
+}
+
+.card__item-name {
+  width: 166px;
```
Comment (silchyn) [inline src/styles/card.css:22]: do you really need it? and what if card gets bigger?

**Review by silchyn — CHANGES_REQUESTED** [review]

File: src/index.html [hunk 13]
```diff
@@ -1,13 +1,70 @@
...
+
+          <li class="stars__star"></li>
+
+          <li class="stars__star"></li>
+
+          <li class="stars__star"></li>
+
+          <li class="stars__star"></li>
+        </ul>
+        <span>Reviews: 5</span>
```
Comment (silchyn) [inline src/index.html:52]: use semantics

**Review by silchyn — APPROVED** [review]

**Comment by Shevtsooov** [comment]
@silchyn I have changed that `span` into `p`.
Is that correct?

**Pull request closed without merging**